### ISSUE TITLE
[ ERROR ] Unexpected error: IOError: [Errno 0] Error

### ISSUE THREAD

**Issue by flywojt** [robotframework/robotframework#2709]
My tests starts to fall down. Everything worked fine until I update my Windows 10 two days ago. I downgrade windows but this doesn't help. I also tried to upgrade robot framework and selenium2library and this doesn't helped. This bug occurs for every browser I used (chrome, ff and edge). 

[ ERROR ] Unexpected error: IOError: [Errno 0] Error
Traceback (most recent call last):
  File "c:\python27\lib\site-packages\robot\utils\application.py", line 83, in _execute
    rc = self.main(arguments, **options)
  File "c:\python27\lib\site-packages\robot\run.py", line 445, in main
    result = suite.run(settings)
  File "c:\python27\lib\site-packages\robot\running\model.py", line 248, in run
    self.visit(runner)
  File "c:\python27\lib\site-packages\robot\model\testsuite.py", line 161, in visit
    visitor.visit_suite(self)
  File "c:\python27\lib\site-packages\robot\model\visitor.py", line 86, in visit_suite
    suite.suites.visit(self)
  File "c:\python27\lib\site-packages\robot\model\itemlist.py", line 76, in visit
    item.visit(visitor)
  File "c:\python27\lib\site-packages\robot\model\testsuite.py", line 161, in visit
    visitor.visit_suite(self)
  File "c:\python27\lib\site-packages\robot\model\visitor.py", line 87, in visit_suite
    suite.tests.visit(self)
  File "c:\python27\lib\site-packages\robot\model\itemlist.py", line 76, in visit
    item.visit(visitor)
  File "c:\python27\lib\site-packages\robot\model\testcase.py", line 74, in visit
    visitor.visit_test(self)
  File "c:\python27\lib\site-packages\robot\running\runner.py", line 159, in visit_test
    self._output.end_test(ModelCombiner(test, result))
  File "c:\python27\lib\site-packages\robot\output\output.py", line 59, in end_test
    LOGGER.end_test(test)
  File "c:\python27\lib\site-packages\robot\output\logger.py", line 183, in end_test
    logger.end_test(test)
  File "c:\python27\lib\site-packages\robot\output\console\verbose.py", line 51, in end_test
    self._writer.status(test.status, clear=True)
  File "c:\python27\lib\site-packages\robot\output\console\verbose.py", line 114, in status
    self._clear_status()
  File "c:\python27\lib\site-packages\robot\output\console\verbose.py", line 124, in _clear_status
    self._write_info()
  File "c:\python27\lib\site-packages\robot\output\console\verbose.py", line 90, in _write_info
    self._stdout.write(self._last_info)
  File "c:\python27\lib\site-packages\robot\output\console\highlighting.py", line 51, in write
    self.stream.write(console_encode(text, stream=self.stream))

**Comment by pekkaklarck** [comment]
I have no idea idea what's causing this but quick googling showed that others have had similar errors. For example, this looks _very_ similar: https://github.com/tartley/colorama/issues/101

**Comment by gonzo60** [comment]
From time to time I have this same problem.

**Comment by bkhouri** [comment]
There is not enough information to determine the root cause.  At a minimum,  we need to know what keyword generated the issue.  Even better would be a test case that reproduces this issue.

**Comment by pekkaklarck** [comment]
This could be relevant: https://stackoverflow.com/questions/11176724/python-file-operations

**Comment by ianlg** [comment]
@flywojt have you found a solution to this issue? I'm having the same problem.

**Comment by ianlg** [comment]
I have just checked my error message is exactly the same as above. I too remember having updated windows 10 then I started to have problems. Does anyone think it has something to do with the win10 update? I tried everything & reinstalled all the components of Robot framework & still no luck.

Noticed the extra text that appear while running the test in pycharm

![capture](https://user-images.githubusercontent.com/22192574/32982623-2aac68ae-cc7f-11e7-9ccf-6dc5850fd4d4.JPG)

**Comment by flywojt** [comment]
@ianlg no, I am not. This must be conected with windows or configuration because on iOS on MAC my tests works fine.

**Comment by ianlg** [comment]
@flywojt how did you manage to fix the issue you mentioned above?

**Comment by pekkaklarck** [comment]
More and more people are reporting this issue when using ChromeDriver also on the Slack channel. Based on the accepted answer to the [StackOverflow question](https://stackoverflow.com/questions/11176724/python-file-operations) I already mentioned, this weird `IOError: [Errno 0]` error occurs on Windows if you read and write to file without using `seek` between. Based on traceback included in the description to this issue, the problem occurs on Robot side when it's writing content to the console. Robot itself never reads from console, but could it be that latest versions of ChromeDriver or Chrome have started doing it?

Although it's unlikely that this problem is caused by Robot, we could possibly workaround it by using `seek` before we write content to console.  Could someone encountering this issue test would adding line

    self.stream.seek(0, 2)

right before [the line causing the error]( https://github.com/robotframework/robotframework/blob/master/src/robot/output/console/highlighting.py#L51) help?

**Comment by pekkaklarck** [comment]
Also, has someone encountered this problem *without* using ChromeDriver? If not, then the real problem clearly is in ChromeDriver or in Chrome itself.

Those encountering the problem with ChromeDriver, do you remember which version started causing it, or could you downgrade to older versions to see when the problem starts appearing? That information could help reporting the problem to ChromeDrivers tracker.

**Comment by ianlg** [comment]
Hi, I also experienced the problem using firefox & edge

**Comment by pekkaklarck** [comment]
Thanks for the commend @ianlg! It seems this isn't only an issue with Chrome(Driver) then. Has anyone encountered this when not using Selenium at all? Does it only occur on Windows? Finally, please test does the [workaround in my comment above](https://github.com/robotframework/robotframework/issues/2709#issuecomment-345646329) help!

**Comment by flywojt** [comment]
This bug occurs for every browser I used (chrome, ff and edge). On Mac and iOS my tests works fine so it looks like that this bug is connected with windows.

**Comment by pekkaklarck** [comment]
Thanks for update @flywojt! Have you encountered the problem when not using Selenium? Can you test updating your SL installation with the workaround I posted?

**Comment by flywojt** [comment]
I haven't tried run tests without selenium. I will try this workaround and give you feedback.

**Comment by Zac4** [comment]
@pekkaklarck The suggested solution doesn't work. File "C:\Python27\lib\site-packages\robot\output\console\highlighting.py", line 52, in write
    self.stream.seek(0, 2)
`IOError: [Errno 22] Invalid argument`. 

Has anyone found a working solution to this annoying issue?

**Comment by pekkaklarck** [comment]
Can you show the full traceback? Have you redirected stdout to a file? Are you running tests locally or on CI?

**Comment by Zac4** [comment]
@pekkaklarck The tests are running locally. Please see full traceback below:
`[ Traceback (most recent call last):
  File "C:\Python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\lib\site-packages\robot\run.py", line 550, in <module>
    run_cli(sys.argv[1:])
  File "C:\Python27\lib\site-packages\robot\run.py", line 489, in run_cli
    return RobotFramework().execute_cli(arguments, exit=exit)
  File "C:\Python27\lib\site-packages\robot\utils\application.py", line 46, in execute_cli
    rc = self._execute(arguments, options)
  File "C:\Python27\lib\site-packages\robot\utils\application.py", line 92, in _execute
    details, rc=FRAMEWORK_ERROR)
  File "C:\Python27\lib\site-packages\robot\utils\application.py", line 106, in _report_error
    self._logger.error(message)
  File "C:\Python27\lib\site-packages\robot\output\loggerhelper.py", line 60, in error
    self.write(msg, 'ERROR')
  File "C:\Python27\lib\site-packages\robot\output\loggerhelper.py", line 63, in write
    self.message(Message(message, level, html))
  File "C:\Python27\lib\site-packages\robot\output\logger.py", line 141, in message
    logger.message(msg)
  File "C:\Python27\lib\site-packages\robot\output\console\verbose.py", line 66, in message
    self._writer.error(msg.message, msg.level, clear=self._running_test)
  File "C:\Python27\lib\site-packages\robot\output\console\verbose.py", line 143, in error
    self._stderr.error(message, level)
  File "C:\Python27\lib\site-packages\robot\output\console\highlighting.py", line 73, in error
    self.write('[ ', flush=False)
  File "C:\Python27\lib\site-packages\robot\output\console\highlighting.py", line 52, in write
    self.stream.seek(0, 2)
IOError: [Errno 22] Invalid argument`

**Comment by Zac4** [comment]
or am i doing something wrong here? 
![image](https://user-images.githubusercontent.com/6584147/33206923-c51959c8-d103-11e7-9d28-434011bc4634.png)
Thank you in advance.

**Comment by pekkaklarck** [comment]
In the above snippet you use `seek(0, 0)` but traceback shows correct `seek(0, 2)`. Hope you can try debugging more.

**Comment by ianlg** [comment]
Hi @pekkaklarck & @Zac4 hope this gets fixed soon. Appreciate your help.

**Comment by pekkaklarck** [comment]
I cannot reproduce this problem and thus cannot do much to fix it. Getting it fixed very highly depends on users encountering debugging it themselves. I have asked has anyone encountered this without using Selenium but nobody has replied anything. I posted a possible workaround by adding `seek(0, 2)`, but only @Zac4 has reportedly tested it. Although the workaround didn't work with @Zac4, it would be good to hear experiences from others.

If you don't know how to apply the workaround, there are instructions:

1. Find file `robot/output/console/highlighting.py` under your Robot Framework installation. You should find it under `C:\Python27\Lib\site-packages` or similar.

2. Change the [write method](https://github.com/robotframework/robotframework/blob/master/src/robot/output/console/highlighting.py#L50) to contain this implementation:

Another thing you could do to help provide more information is adding line `open('debug.output', 'w').write(str(self.stream) + '\n')` to the beginning of the aforementioned `write` function, and telling what is written to the generated `debug.output` file.

**Comment by pekkaklarck** [comment]
Some more questions to the users encountering this issue based on the new information in the vscode issue linked above:

1. Which terminal do you use? The old command prompt, PowerShell, vscode, Git Bash, Cygwin, or something else?

2. Does the problem go away if you disable console colors by using `--consolecolors off` when starting test execution?

**Comment by ianlg** [comment]
Hi,

I've done what you said:

    def _get_highlighter(self, stream, colors):
        options = {'AUTO': Highlighter if isatty(stream) else NoHighlighting,
                   'ON': Highlighter,
                   'OFF': NoHighlighting,
                   'ANSI': AnsiHighlighter}
        try:
            highlighter = options[colors.upper()]
        except KeyError:
            raise DataError("Invalid console color value '%s'. Available "
                            "'AUTO', 'ON', 'OFF' and 'ANSI'." % colors)
        return highlighter(stream)

    def write(self, text, flush=True):
        with open('debug.output', 'a') as f:
            f.write(str(self.stream) + '\n')
        self.stream.seek(0, 2)
        self.stream.write(console_encode(text, stream=self.stream))
        if flush:
            self.flush()

    def flush(self):
        self.stream.flush()

    def highlight(self, text, status=None, flush=True):
        if self._must_flush_before_and_after_highlighting():
            self.flush()
            flush = True
        with self._highlighting(status or text):
            self.write(text, flush)



but it gave the the error straight away when I ran the test.

**Comment by pekkaklarck** [comment]
Which error? What was written to the 'debug.output' file?

**Comment by ianlg** [comment]
HI,

Here is the error in pycharm

Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "c:\python27\lib\site-packages\robot\run.py", line 550, in <module>
    run_cli(sys.argv[1:])
  File "c:\python27\lib\site-packages\robot\run.py", line 489, in run_cli
    return RobotFramework().execute_cli(arguments, exit=exit)
  File "c:\python27\lib\site-packages\robot\utils\application.py", line 46, in execute_cli
    rc = self._execute(arguments, options)
  File "c:\python27\lib\site-packages\robot\utils\application.py", line 92, in _execute
    details, rc=FRAMEWORK_ERROR)
  File "c:\python27\lib\site-packages\robot\utils\application.py", line 106, in _report_error
    self._logger.error(message)
  File "c:\python27\lib\site-packages\robot\output\loggerhelper.py", line 60, in error
    self.write(msg, 'ERROR')
  File "c:\python27\lib\site-packages\robot\output\loggerhelper.py", line 63, in write
    self.message(Message(message, level, html))
  File "c:\python27\lib\site-packages\robot\output\logger.py", line 141, in message
    logger.message(msg)
  File "c:\python27\lib\site-packages\robot\output\console\verbose.py", line 66, in message
    self._writer.error(msg.message, msg.level, clear=self._running_test)
  File "c:\python27\lib\site-packages\robot\output\console\verbose.py", line 143, in error
    self._stderr.error(message, level)
  File "c:\python27\lib\site-packages\robot\output\console\highlighting.py", line 75, in error
    self.write('[ ', flush=False)
  File "c:\python27\lib\site-packages\robot\output\console\highlighting.py", line 53, in write
    self.stream.seek(0, 2)
IOError: [Errno 22] Invalid argument


Which directory can I find debug.output file?

**Comment by Zac4** [comment]
@ianlg the debug.output file should be automatically created within your IDE. The file gets created immediately you execute the command. 
![image](https://user-images.githubusercontent.com/6584147/33272684-cab29440-d382-11e7-8d42-866bb281b578.png)

**Comment by Zac4** [comment]
@ianlg try this, and let us know if it fixes it for you. Try number one first, then if doesn't work, try number two:
1. `self.stream.seek(0, 0)`  if fails try no 2
2. `self.stream.seek(0)`

**Comment by ianlg** [comment]
HI @Zac4 I tried both and they work fine, I didn't see any errors apart from the weird line of text that I mentioned above.

**Comment by pekkaklarck** [comment]
Some notes related to questions and comments above:

1. The `debug.output` file is created to the directory where tests are executed from. Thanks @ianlg for posting its content. You can remove the related part from the code now.

2. @Zac4 @ianlg: Did `self.stream.seek(0)` fix the problem for you? Very interesting. That one moves the current position into the beginning of the file when `seek(0, 2)` moves it at the end. Moving to the beginning likely means that produced console output is messed up. I also noticed that when using stdout or stderr, `seek` only works if output is redirected to a file. This means that `seek` alone isn't a suitable general workaround.

3. Same/similar `Errno 0` problem reported as Microsoft/vscode#36630 seems to only occur only for users who have Windows 10 Fall Creators Update installed. Is that the same with Robot Framework users encountering the issue? It could be that the problem is caused by changes to Windows itself.

4. I have earlier asked users encountering this issue to the following questions. If you want to help this issue getting fixed, please answer!

    - Has anyone encountered this when not using Selenium at all?

    - Which terminal do you use? The old command prompt, PowerShell, vscode, Git Bash, Cygwin, or something else?

    - Does the problem go away if you disable console colors by using `--consolecolors` off when starting test execution?

**Comment by Zac4** [comment]
@pekkaklarck `self.stream.seek(0)` seems to have fixed the issue. Although it's not a pretty workaround

Also, @ianlg i am not sure if @pekkaklarck will find a solution to that extra DevTools that keeps using random port on each test case at the time of execution. but i know for definite that is annoyingly caused by recent Windows 10 updates.

**Comment by pekkaklarck** [comment]
That DevTools thing is unlikely to have anything to do with this issue. It's probably a message logged by the Selenium browser driver you use.

Could someone reliably reproducing this test does the problem occur also without using Selenium? Also, could someone test does `--consolecolors off` fix the problem? Especially the latter would be really important information as it would both provide a simple workaround (no changes to code) and provide info related to where the actual problem is. Apparently handling console colors have changed in latest Windows versions.

**Comment by miniksa** [comment]
As someone who works on the console in Windows, we're not actually aware of anything changing in regards to how colors are handled by external applications interfacing with the console subsystem. If you can help us identify which API call isn't working as it used to, we can investigate and work on a fix. Compatibility is very important to us. We're just having a hard time navigating the many levels of abstraction that are present in these scenarios to find the actual Win32 API call at the bottom that is acting differently.

**Comment by ianlg** [comment]
I ran another test run this time self.stream.seek(0) failed. Now trying self.stream.seek(0, 0)

**Comment by ianlg** [comment]
self.stream.seek(0, 0) resulted in a failure as well =(

**Comment by LexiPham** [comment]
I can't run my tests anymore because this error happened  frequently. Any fixing for this, anything news

**Comment by LexiPham** [comment]
Which terminal do you use? The old command prompt, PowerShell, vscode, Git Bash, Cygwin, or something else? --> I'm using Pycharm

Does the problem go away if you disable console colors by using --consolecolors off when starting test execution? --> no

**Comment by LexiPham** [comment]
Hi,
 
I did this:

Find file robot/output/console/highlighting.py under your Robot Framework installation. You should find it under C:\Python27\Lib\site-packages or similar.

Change the write method to contain this implementation:

def write(self, text, flush=True):
  with open('debug.output', 'a') as f:
      f.write(str(self.stream) + '\n')
  self.stream.seek(0, 2)
  self.stream.write(console_encode(text, stream=self.stream))
  if flush:
      self.flush()
Report does the workaround work.

Tell what's written to the generated debug.output file.

[1209/215700.260:ERROR:process_reader_win.cc(123)] NtOpenThread: {Access Denied} A process has requested access to an object, but has not been granted those access rights.  (0xc0000022)
[1209/215700.359:ERROR:exception_snapshot_win.cc(87)] thread ID 7836 not found in process
[1209/215700.359:WARNING:crash_report_exception_handler.cc(62)] ProcessSnapshotWin::Initialize failed

In the Pycharm console:

Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "c:\python27\lib\site-packages\robot\run.py", line 550, in <module>
    run_cli(sys.argv[1:])
  File "c:\python27\lib\site-packages\robot\run.py", line 489, in run_cli
    return RobotFramework().execute_cli(arguments, exit=exit)
  File "c:\python27\lib\site-packages\robot\utils\application.py", line 46, in execute_cli
    rc = self._execute(arguments, options)
  File "c:\python27\lib\site-packages\robot\utils\application.py", line 92, in _execute
    details, rc=FRAMEWORK_ERROR)
  File "c:\python27\lib\site-packages\robot\utils\application.py", line 106, in _report_error
    self._logger.error(message)
  File "c:\python27\lib\site-packages\robot\output\loggerhelper.py", line 60, in error
    self.write(msg, 'ERROR')
  File "c:\python27\lib\site-packages\robot\output\loggerhelper.py", line 63, in write
    self.message(Message(message, level, html))
  File "c:\python27\lib\site-packages\robot\output\logger.py", line 141, in message
    logger.message(msg)
  File "c:\python27\lib\site-packages\robot\output\console\verbose.py", line 66, in message
    self._writer.error(msg.message, msg.level, clear=self._running_test)
  File "c:\python27\lib\site-packages\robot\output\console\verbose.py", line 143, in error
    self._stderr.error(message, level)
  File "c:\python27\lib\site-packages\robot\output\console\highlighting.py", line 75, in error
    self.write('[ ', flush=False)
AttributeError: 'HighlightingStream' object has no attribute 'write'

**Comment by pekkaklarck** [comment]
This problem is most likely caused by the same change to Windows console that was behind Microsoft/vscode#36630. The root cause to that issue is fixed, and apparently it should be in Windows Insider builds early next year. I don't know when normal users will get it, and obviously there's even a change that our bug is different.

It seems that at least @LexiPham and @ianlg have encountered this issue when using PyCharm. Could you try running tests using the normal Windows console instead?

**Comment by pekkaklarck** [comment]
I finally saw this problem myself when organizing a training in a classroom with machines having Windows 10 machines. In my tests it only occurred when using PyCharm terminal (probably would have also occurred using VSCode) and not when using the standard Windows command prompt. Olso on PyCharm it didn't occur always but annoyingly frequently (about every fourth time). Is this what others have noticed too?

I was also able to create  a workaround for the problem and committed it in 828c67695d85519e4435c556c43ed1b00985df05. It would be great if users encountering this error could test the workaround. The easiest way to get it is installing the latest code directly using pip:

    pip install --upgrade https://github.com/robotframework/robotframework/zipball/master

**Comment by LexiPham** [comment]
Hi @pekkaklarck ,

It works for me, don't see the error anymore.

It has some weird lines that I haven't seen while running my tests before, but it didn't affect  the test results so far.

[10788:15320:1220/183512.412:ERROR:process_metrics.cc(105)] NOT IMPLEMENTED
[10788:15320:1220/183512.412:ERROR:process_metrics.cc(105)] NOT IMPLEMENTED
[10788:15320:1220/183512.412:ERROR:process_metrics.cc(105)] NOT IMPLEMENTED

Thank you,

**Comment by pekkaklarck** [comment]
Great that the workaround works!

The weird lines you get on the console are most likely caused by the Selenium driver or browser you use. I've seen at least ChromeDriver/Chrome combo emitting such noise on Windows sometimes.

**Comment by pekkaklarck** [comment]
I guess we can consider this issue resolved. Because the problem doesn't occur when using the normal command prompt, I don't consider this important enough to warrant an immediate bug fix release. If you need to get the fix ASAP, see my comment a bit above how to install the latest code using pip.

**Comment by shi-kk** [comment]
Hi! I have run into this problem this week.

I'm using windows 10
Running test in Pycharm, using behave and there is NO selenium involved (service tests)
My tests run OK in Windows console and throw the "IOError Errno 0" in Pycharm console

**Comment by pekkaklarck** [comment]
@shi-kk the bug itself is in the latest Windows versions. It's apparently already fixed but it takes some time before the fix is rolled out. For more details see Microsoft/vscode#36630.

For Robot Framework I've implemented a workaround that avoids the problem in commit 828c67695d85519e4435c556c43ed1b00985df05. Something similar could be done with Behave as well.

**Comment by jbub** [comment]
So we have tried running with the commit 828c676 and seems like its not helping on Windows 7 Enterprise, here is a traceback:

```
[ ERROR ] Unexpected error: IOError: [Errno 22] Invalid argument
Traceback (most recent call last):
  File "c:\mylocation\src\robotframework\src\robot\utils\application.py", line 83, in _execute
    rc = self.main(arguments, **options)
  File "c:\mylocation\src\robotframework\src\robot\run.py", line 445, in main
    result = suite.run(settings)
  File "c:\mylocation\src\robotframework\src\robot\running\model.py", line 248, in run
    self.visit(runner)
  File "c:\mylocation\src\robotframework\src\robot\model\testsuite.py", line 161, in visit
    visitor.visit_suite(self)
  File "c:\mylocation\src\robotframework\src\robot\model\visitor.py", line 86, in visit_suite
    suite.suites.visit(self)
  File "c:\mylocation\src\robotframework\src\robot\model\itemlist.py", line 76, in visit
    item.visit(visitor)
  File "c:\mylocation\src\robotframework\src\robot\model\testsuite.py", line 161, in visit
    visitor.visit_suite(self)
  File "c:\mylocation\src\robotframework\src\robot\model\visitor.py", line 86, in visit_suite
    suite.suites.visit(self)
  File "c:\mylocation\src\robotframework\src\robot\model\itemlist.py", line 76, in visit
    item.visit(visitor)
  File "c:\mylocation\src\robotframework\src\robot\model\testsuite.py", line 161, in visit
    visitor.visit_suite(self)
  File "c:\mylocation\src\robotframework\src\robot\model\visitor.py", line 86, in visit_suite
    suite.suites.visit(self)
  File "c:\mylocation\src\robotframework\src\robot\model\itemlist.py", line 76, in visit
    item.visit(visitor)
  File "c:\mylocation\src\robotframework\src\robot\model\testsuite.py", line 161, in visit
    visitor.visit_suite(self)
  File "c:\mylocation\src\robotframework\src\robot\model\visitor.py", line 87, in visit_suite
    suite.tests.visit(self)
  File "c:\mylocation\src\robotframework\src\robot\model\itemlist.py", line 76, in visit
    item.visit(visitor)
  File "c:\mylocation\src\robotframework\src\robot\model\testcase.py", line 74, in visit
    visitor.visit_test(self)
  File "c:\mylocation\src\robotframework\src\robot\running\runner.py", line 159, in visit_test
    self._output.end_test(ModelCombiner(test, result))
  File "c:\mylocation\src\robotframework\src\robot\output\output.py", line 59, in end_test
    LOGGER.end_test(test)
  File "c:\mylocation\src\robotframework\src\robot\output\logger.py", line 183, in end_test
    logger.end_test(test)
  File "c:\mylocation\src\robotframework\src\robot\output\console\verbose.py", line 51, in end_test
    self._writer.status(test.status, clear=True)
  File "c:\mylocation\src\robotframework\src\robot\output\console\verbose.py", line 117, in status
    self._stdout.write(' |\n')
  File "c:\mylocation\src\robotframework\src\robot\output\console\highlighting.py", line 53, in write
    self.flush()
  File "c:\mylocation\src\robotframework\src\robot\output\console\highlighting.py", line 66, in flush
    self.stream.flush()
```

**Comment by pekkaklarck** [comment]
@jbub You are encountering some other problem. This issue only occurs with latest Windows 10 versions.

**Comment by magork** [comment]
Hy @pekkaklarck,
I had the same problem on a virtual machine which runs win 10 but with 
pip install --upgrade https://github.com/robotframework/robotframework/zipball/master
resolved for me also. So I can say also that the workaround works!

**Comment by pekkaklarck** [comment]
Great that the fix works! It was included in RF 3.0.3, so nowadays the normal `pip install --upgrade robotframework` is enough.
